polyphemus/lib/etls/ipi/ipi_etls.rb

```diff
@@ -11,13 +11,74 @@
 
 class Polyphemus
   module Ipi
+
+    module IpiRecordNames
+      def corrected_record_name(record_name)
+        if record_name.include?('POOL')
+          record_name.gsub(/_/, '.').sub(/^([^.]*\.[^.]*\.)(.*)$/) do
+            first = $1
+            second = $2
+            first + second.downcase
+          end
+        else
+          record_name.sub(/^([^_.]*[_.][^_.]*[_.])([^_.]*)(.*)$/) do
+            first = $1
+            second = $2
+            third = $3
+            first.gsub(/_/, '.') + second.downcase + third
+          end
+        end
+      end
+
+
+      SINGLE_CELL_ROOT = /^single_cell_[^\/]*/
+      POOL_CONTAINER = /.*\/IPIPOOL[^\/]+(\/[^\/]*)?$/
+      NON_POOLED_CONTAINER = /.*\/IPI((?!POOL)[^\/])+(\/[^\/]*)?$/
+    end
+
+    module RawFastqLinkers
```

Comment:
Super nit: I think there are also raw fastq files for other (non-single-cell) assays, like whole exome sequencing and bulk RNA sequencing. Not sure if here you want to specify this module as `SingleCellRawFastqLinkers`?